docs/observability.md

```diff
@@ -0,0 +1,84 @@
+
+## Prerequisites
+
+Ensure the following tools are installed on your local machine:
+- **Kubernetes**: A running Kubernetes cluster.
+- **kubectl**: To manage the cluster.
+- **Helm**: To install and manage Helm charts for Prometheus and Grafana.
+
+You should also have access to the Kubernetes cluster's kubeconfig file (`test-cluster-kubeconfig.yaml`), which will be used for running the make target.
+
+## Steps to Install the Grafana Dashboard
+
+### 1. Build and Set Up the Cluster (Optional)
+If you haven’t already set up your Kubernetes cluster with the necessary CSI driver and Prometheus metrics services, you can do so by running the following command:
+```bash
+make mgmt-and-capl-cluster
+```
+This command creates a management cluster and CAPL (Cluster API for Linode) cluster, installs the Linode CSI driver, and applies the necessary configurations to expose the CSI metrics.
+
+### 2. Run the Grafana Dashboard Setup
+The make target `grafana-dashboard` will install Prometheus and Grafana in a Kubernetes namespace (`monitoring` by default) and configure a Grafana dashboard from a local JSON file. To execute this setup, run:
+
+```bash
+make grafana-dashboard
+```
+
+### 3. What Happens During the Setup?
+
+- **Node Scheduling**: The script checks whether worker nodes exist in the cluster. If no worker nodes are found, it untaints the control-plane nodes to allow Prometheus and Grafana to be scheduled.
+
+- **Helm Repository Setup**: Helm repositories for Prometheus and Grafana are added if not already present.
+
+- **Prometheus Installation**: The `prometheus` Helm chart is installed (or upgraded) in the monitoring namespace. It scrapes metrics from the CSI driver and other services.
+
+- **Grafana Installation**: The `grafana` Helm chart is installed (or upgraded) in the same namespace, with Prometheus set as the data source.
+
+- **Dashboard ConfigMap**: The script uploads the Grafana dashboard configuration stored locally in `docs/dashboard.json` into a Kubernetes ConfigMap, ensuring the dashboard is automatically loaded into Grafana.
+
+- **Port Forwarding**: The script port-forwards the Grafana service to `localhost:3000`, allowing you to access the Grafana dashboard on your local machine.
+
+### 4. Accessing the Grafana Dashboard
+
+Once the setup is complete, open your web browser and navigate to:
+```
+http://localhost:3000
+```
+
+Log in using the following credentials:
+- Username: `admin`
+- Password: `admin`
+
+These details can be customized via the `install-monitoring-tools.sh` script if needed. 
+
+To understand the graphs and the metrics, go through the [Metrics Documentation](metrics-documentation.md).
+
+### 5. Stopping the Port Forwarding
+
+To stop the Grafana port forwarding, run:
+```bash
+kill <PID>
+```
+Replace `<PID>` with the process ID provided by the script during the setup.
+
```

Comment:
Could you add a command to show user how to get the exact PID? Incase they don't have access to the script output

Reply:
Sure